lib/stripe_mock/request_handlers/checkout_session.rb

```diff
@@ -3,9 +3,88 @@ module RequestHandlers
+          payment_intent = nil
+          setup_intent = nil
+          case params[:mode]
+          when nil, "payment"
+            require_params(:line_items) if params[:line_items].nil? || params[:line_items].empty?
+            payment_intent = new_payment_intent(nil, nil, {
+              amount: amount,
+              currency: currency,
+              customer: params[:customer],
+              line_items: params[:line_items],
+              payment_method_options: params[:payment_method_options],
+              payment_method_types: params[:payment_method_types]
+            }.merge(params[:payment_intent_data] || {}), nil)[:id]
+          when "setup"
+            if !params[:line_items].nil? && !params[:line_items].empty?
+              throw Stripe::InvalidRequestError.new("You cannot pass `line_items` in `setup` mode", :line_items, http_status: 400)
+            end
+            setup_intent = new_setup_intent(nil, nil, {
+              customer: params[:customer],
+              payment_method_options: params[:payment_method_options],
+              payment_method_types: params[:payment_method_types]
+            }.merge(params[:setup_intent_data] || {}), nil)[:id]
+            payment_status = "no_payment_required"
+          when "subscription"
+            require_params(:line_items) if line_items.nil? || line_items.empty?
+            # TODO: Stripe does not create the Subscription when creating the Session, add support for a way to create
+            # a subscription using this session.
+          else
+            throw Stripe::InvalidRequestError.new("Invalid mode: must be one of payment, setup, or subscription", :mode, http_status: 400)
+          end
+
+          checkout_sessions[id] = {
+            id: id,
+            object: "checkout.session",
+            allow_promotion_codes: nil,
+            amount_subtotal: amount,
+            amount_total: amount,
+            automatic_tax: {
+              enabled: false,
+              status: nil
+            },
+            billing_address_collection: nil,
+            cancel_url: "http://localhost:3000/connect/stold4/pay_later",
+            client_reference_id: nil,
+            currency: currency,
+            customer: nil,
+            customer_details: nil,
+            customer_email: nil,
+            livemode: false,
+            locale: nil,
+            metadata: {},
+            mode: "setup",
+            payment_intent: payment_intent,
+            payment_method_options: {},
+            payment_method_types: ["bacs_debit", "card"],
+            payment_status: payment_status,
+            setup_intent: setup_intent,
+            shipping: nil,
+            shipping_address_collection: nil,
+            submit_type: nil,
+            subscription: nil,
+            success_url: "http://localhost:3000/connect/stold4/pay_later_callback?session_id={CHECKOUT_SESSION_ID}&date=2021-08-02",
+            total_details: nil,
+            url: nil
```

Comment:
It would be nice to have at least a fake url, like 
```suggestion
            url: "https://checkout.stripe.com/pay/cs_test_abc123"
```